### PR TITLE
feat!: refactor repo to use composition 

### PR DESCRIPTION
**Pull request by sharanyavinod**
## Description

- Refactored repository to use composition instead of inheritance
- Fix security vulnerabilities and remove unnecessary dependencies
- New prop `removeDefaultStyles` added for removing default AEM styles/className if needed. 
- New prop `model` added to support passing in a model object if prefetched
- New prop `components` added to support component mapping without using MapTo
- Introduced new wrapper `EditableComponent` for enabling AEM integration which removes the need for `withModel`,`withMappable` and `withComponentMappingContext `
- New `fetchModel` method exposed for simple fetch of model - helpful in SSR/Nextjs scenarios 
- Project structure updated
- Updated and detailed readme added
- Setup linting rules as per editor conventions 

Suggestions for alternative names for props introduced above are welcome.

## Motivation and Context

- Simpler and performant code
- Updated to keep in-line with features introduced by React over time
- Better support for SSR

## How Has This Been Tested?

- Existing tests updated and new tests added

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,86 +16,95 @@
     "url": "https://github.com/adobe/aem-react-editable-components/issues"
```
Comment (blackbyte-pl) [inline package.json:16]: I guess with this change we need to change a major version of this repo.

File: src/utils/Utils.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+   * @param item - the item data
+   * @returns the transformed data
+   */
+  modelToProps(item: Model): ModelProps {
+    if (!item || !Object.keys(item).length) {
+      return { cqPath: '' };
+    }
+
+    const keys = Object.getOwnPropertyNames(item);
+    const props: any = {};
```
Comment (blackbyte-pl) [inline src/utils/Utils.ts:51]: if any has to be here, then please add tsignore comment

**Review by blackbyte-pl — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (sharanyavinod) [reply package.json:16]: Indeed, this will be v2 as there are quite a few breaking changes

**Review by grubyak — APPROVED** [review]
great refactoring! 👏 🥇

**Comment by mitchross** [comment]
Where is the migration guide for this? This breaks alot of stuff especially around extending Responsive Grid, yet there is no migration guide.

**Pull request merged**